setup.cfg

```diff
@@ -26,6 +26,7 @@ classifiers =
 zip_safe = False
 python_requires = >=3.7
 setup_requires =
+    numpy
```

Comment:
This is not really needed here, because by the time `setup` is run, numpy needs to be installed already.  Instead you need to list it in pyproject.toml, and use `oldest-supported-numpy` instead, as that's what will be used to build the C code.

See

https://github.com/astropy/astropy/blob/5f8866e41164293e463a3cc53fadb5566ba488cc/pyproject.toml#L7

for an example.